python-package/lightgbm/basic.py

```diff
@@ -1776,8 +1799,10 @@ def construct(self):
         if self.handle is None:
             if self.reference is not None:
                 reference_params = self.reference.get_params()
-                if self.get_params() != reference_params:
-                    _log_warning('Overriding the parameters from Reference Dataset.')
+                params = self.get_params()
+                if params != reference_params:
+                    if self._compare_params_for_warning(params, reference_params) is False:
```

Comment:
According to `get_params`, both `params` and `reference_params` can be `None`. In this case they cannot be directly iterated in the `_compare_params_for_warning` method.

Reply:
Thanks shiyu. I also found this issue in the CI. Will add a check on this.